### PR TITLE
[ic-date-input/ic-date-picker]: Add `hideLabel` functionality

### PR DESCRIPTION
**Pull request by jd239**
## Summary of the changes
Introduced the 'hideLabel' functionality for date picker and date input.
Changes include:
- Updating labelledby method on `ic-input-component-container`
- Added Cypress test on Date Input
- Regenerated docs & canary docs
- Updated stories
- Added hideLabel to `ic-input-label`
- Updated `ic-input-label` unit tests

## Related issue
N/A

### General 

- [x] Changes to docs package checked and committed.

### Testing

- [x] Relevant unit/component tests and visual regression tests added. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Text resized to 200% with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] All prop combinations work without issue.

### CODE REVIEW

File: packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx [hunk 1]
```diff
@@ -173,6 +173,11 @@ export class DateInput {
    */
   @Prop() hideHelperText: boolean = false;
 
+  /**
+   * If `true`, the label and helper text will be visually hidden, but the required label will still be read out by screen readers.
```
Comment (ad9242) [inline packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx:177]: does this prop need to hide the helper text? there is already a hideHelperText prop

**Review by cd3859 — APPROVED** [review]

File: packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx (hunk 1 above)
Comment (GCHQ-Developer-299) [reply packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx:177]: feels intuitive to me that if you want a labelless field you'd want helper text hidden too - but as a design system i'd agree that we could offer more flexibility if this prop only hides the label.

Comment (ad9242) [reply packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx:177]: there is precedent for hideLabel also hiding the helpertext with ic-text-field. The issue is hideHelperText has already been implemented as another feature request on the date input, so this causes a bit of an inconsistency in the components

Comment (GCHQ-Developer-299) [reply packages/canary-web-components/src/components/ic-date-input/ic-date-input.tsx:177]: Is this not a blocker then, if hideLabel is doing the same thing as in ic-text-field?  I'm happy to remove my Request Changes if you're okay with Date Picker having a partly redundant hideHelperText prop :)

**Review by GCHQ-Developer-299 — APPROVED** [review]

**Review by ad9242 — APPROVED** [review]

**Pull request merged**